### PR TITLE
Add descriptions to all commands

### PR DESCRIPTION
**Pull request by JorikSchellekens**
Addresses #825

### CODE REVIEW

File: src/cli.ts [hunk 1]
```diff
@@ -61,24 +61,31 @@ export const program = new Command();
...
-  .option('--no-result')
-  .option('--no-stubs')
-  .option('--no-strict')
+  .option(
+    '-d, --debug-info',
+    'Include debug information in the compiled bytecode produced by --compile-cario',
+    false,
+  )
+  .option('--print-trees', 'Debug: Print all the intermediate ASTs')
+  .option('--no-stubs', 'Debug: Hide the stubs in the intermedidate ASTs when using print-trees')
```
Comment (JorikSchellekens) [inline src/cli.ts]:
```suggestion
  .option('--no-stubs', 'Debug: Hide the stubs in the intermedidate ASTs when using --print-trees')
```

Nit, can you do this for the rest too?

File: src/cli.ts [hunk 2]
```diff
@@ -61,24 +61,31 @@ export const program = new Command();
...
+  .option('--order <passOrder>', 'Use a custom set of transpilation passes')
   .option('-o, --output-dir <path>', 'Output directory for transpiled Cairo files.', 'warp_output')
-  .option('-d, --debug-info', 'Include debug information.', false)
-  .option('--print-trees')
-  .option('--no-result')
-  .option('--no-stubs')
-  .option('--no-strict')
+  .option(
+    '-d, --debug-info',
+    'Include debug information in the compiled bytecode produced by --compile-cario',
```
Comment (piwonskp) [inline src/cli.ts]: cairo*

File: src/cli.ts [hunk 3]
```diff
@@ -135,19 +142,24 @@ export function runTranspile(files: string[], options: CliOptions) {
...
+  .option(
+    '--highlight <ids...>',
+    'Debug: highlight selected ids in the AST printed by --print-trees',
+  )
+  .option('--order <passOrder>', 'Use a custom set of transpilation passes')
+  .option('-o, --output-dir <path>', 'Output directory for transformed Solidity files')
+  .option('--print-trees', 'Debug: Print all the intermediate ASTs')
+  .option('--no-stubs', 'Debug: Hide the stubs in the intermedidate ASTs when using --print-trees')
+  .option('--no-strict', 'Debug: Allow silent failure of AST consistency checks')
+  .option('--until <pass>', 'Pass to transform to')
```
Comment (piwonskp) [inline src/cli.ts]: Imo "Stop processing at specified pass" would be more straightforward

File: src/cli.ts [hunk 4]
```diff
@@ -200,7 +214,8 @@ export interface IOptionalNetwork {
 
 program
   .command('status <tx_hash>')
-  .option('--network <network>', 'Starknet network URL.', process.env.STARKNET_NETWORK)
+  .description('Get the satus of a transaction')
```
Comment (piwonskp) [inline src/cli.ts]: status*

File: src/cli.ts [hunk 5]
```diff
@@ -226,14 +242,20 @@ export interface SolcInterfaceGenOptions {
 
 program
   .command('gen_interface <file>')
+  .description(
+    'Use native Cario conracts in your Soldity by creating a Solidity interface and a Cairo translation contract for the target Cairo contract',
```
Comment (piwonskp) [inline src/cli.ts]: contracts*

Comment (piwonskp) [reply src/cli.ts]: Cairo*

File: src/cli.ts [hunk 6]
```diff
@@ -296,44 +320,46 @@ export type ICallOrInvokeProps = ICallOrInvokeProps_ &
 
 program
   .command('invoke <file>')
-  .requiredOption('--address <address>', 'Address of contract to invoke.')
-  .requiredOption('--function <function>', 'Function to invoke.')
+  .description('Invoke a function on a warped contract using the Solidity abi')
```
Comment (piwonskp) [inline src/cli.ts:371]: As there is an option  `--use-cairo-abi` it should be `Uses Solidity abi by default`

Comment (swapnilraj) [reply src/cli.ts:371]: I don't think we should be advertising the --use-cairo-abi to the user; its really for debugging purposes.
If somehow we can remove it from it a production build that would be ideal IMO

**Comment by swapnilraj** [comment]
Hey @piwonskp thanks for your thorough review, really appreciate it.
For the future can we think of a way to run grammar check on ts files that produce strings that the user can see.
Let's make a task for it and mark it good first issue + help wanted

**Review by piwonskp — APPROVED** [review]

File: src/cli.ts [hunk 7]
```diff
@@ -61,24 +61,31 @@ export const program = new Command();
...
+    false,
+  )
+  .option('--print-trees', 'Debug: Print all the intermediate ASTs')
+  .option('--no-stubs', 'Debug: Hide the stubs in the intermedidate ASTs when using --print-trees')
+  .option('--no-strict', 'Debug: Allow silent failure of AST consistency checks')
   .option('--until <pass>', 'Stops transpilation after the specified pass')
-  .option('--no-warnings')
-  .option('--include-paths <paths...>')
-  .option('--base-path <path>')
+  .option('--no-warnings', 'Suppress printed warnings')
```
Comment (rodrodros) [inline src/cli.ts]: These warnings refers to the warnings given by the solidity compiler and not the transpiler. I would update it to something like "Suppress solc warnings". Or change the code to suppress Warp warnings as well ...

File: src/cli.ts [hunk 8]
```diff
@@ -135,19 +142,24 @@ export function runTranspile(files: string[], options: CliOptions) {
...
+  .description(
+    'Debug tool which applies any set of passes to the AST and writes out the transformed Solidity',
+  )
   .option('--no-compile-errors')
-  .option('--check-trees')
-  .option('--highlight <ids...>')
-  .option('--order <passOrder>')
-  .option('-o, --output-dir <path>')
-  .option('--print-trees')
-  .option('--no-result')
```
Comment (cicr99) [inline src/cli.ts:145]: What about this option?

Comment (swapnilraj) [reply src/cli.ts:145]: On the right is the new ones with the added descriptions

Comment (cicr99) [reply src/cli.ts:145]: Yes, I just meant that `--no-result` option was removed, no description was provided for it either, so I wanted to know if this was something that was intended

Comment (JorikSchellekens) [reply src/cli.ts:145]: I think this was a mistake in https://github.com/NethermindEth/warp/pull/827/commits/1e94ea68bcf66036e2907ed313817f3323da2b98

Comment (JorikSchellekens) [reply src/cli.ts:145]: Correction, this option didn't do anything anymore. It has been removed in https://github.com/NethermindEth/warp/commit/1e94ea68bcf66036e2907ed313817f3323da2b98

**Review by cicr99 — APPROVED** [review]

**Review by GregVardz — APPROVED** [review]

**Pull request merged**